### DISCUSSION TITLE
Releases lifecycle

### DISCUSSION THREAD

**Discussion by maratsal** [okd-project/okd#1401] [General]
Does anyone know how OKD release lifecycle looks like? Is it tied to official Red Hat OpenShift lifecycle? I am curious because of security patches and bug fixes application.

I was looking something similar to this Red Hat page - https://access.redhat.com/support/policy/updates/openshift#overview

**Comment by vrutkovs** [comment]
>Does anyone know how OKD release lifecycle looks like?

OKD is a community distro and it comes without any official support, just community support. As a result, lifecycle is defined by the community's willingness to help others solve issues.

However, we're also limited by one upgrade channel restriction too, so we can't release fixes for OKD 4.10 once 4.11 is in stable.

**Reply by darren-oxford** [reply]
@vrutkovs Out of interest, what is it that limits us to one upgrade channel?

The reason I ask, is that it sometimes causes a problem For example when we had the 4.10 update, it included an issue with CephFS that prevented us from upgrading from 4.9, but there were critical security updates for FCOS that we couldn't get due to the release of 4.10. Essentially we had the choice of insecure or broken.

This particular issue will be mitigated somewhat by CoreOS layering, but it would be interesting to see if we could change things to have multiple upgrade channels, so that the community could apply hot fix changes to previous versions as part of the new build set-up?

**Reply by vrutkovs** [reply]
Its kind of self-imposed, we could have stable-4.10 channel I guess, but we already overdue on a lot of issues here, so lets not make it worse.

**Comment by maratsal** [comment]
Thanks for the input. I am a bit new to the process, but how release process looks like? Is OKD somehow gets Red Hat version of the code, or is it some kind of cherry picking features/bugfixes from there or is main development is done within OKD and then ported to Red Hat version? If there is a clear description of the process somewhere it would be great if you can point me to it.

I just was curious how much effort it would be to keep couple of stable channels up to date with recent bug fixes and security patches.

**Reply by vrutkovs** [reply]
>how release process looks like

Its described at https://hackmd.io/YJBn04R5TDi5Sm9XbOGwZA

>Is OKD somehow gets Red Hat version of the code, or is it some kind of cherry picking features/bugfixes from there or is main development is done within OKD and then ported to Red Hat version?

Most images comprising OCP are being copied to OKD namespace. Some images need tweaking - i.e. installer, marketplace-operator, machine-config-operator - these build OKD-specific version on each PR and get promoted simultaneously. Some images - branding, machine-os-content - are OKD-specific, they never land in OCP.
Once the image is promoted during a pull-request merge release-controller is building a new nightly. Every two weeks (roughly) we pick a nightly and promote it to become OKD stable release.

We still nightlies for every release stream, so maintaining several OKD stream would require:
* additional testing to make sure nightlies are indeed stable
* design channel naming, deprecation procedure, documentation and code changes in release-controller if necessary

**Comment by yuanlinios** [comment]
Add some questions related to OKD release cycle here

1. As 4.11 is already available in 4-stable, there will be no update for the 4.10 version in the 4-stable channel. If we want to apply some fix to the existing 4.10 (which is already the latest in 4-stable), we have to uprade to 4.11, or resort to the 4.10 nightly builds?

2. Is "EUS release" applicable to OKD? I did't remeber it is available when I upgrade 4.8 -> 4.9 -> 4.10
https://docs.okd.io/4.10/updating/understanding-openshift-updates.html
> Extended Update Support
All post-4.7 even-numbered minor releases are labeled as Extended Update Support (EUS) releases. These releases introduce a verified update path between EUS releases, permitting customers to streamline updates of worker worker nodes and formulate update strategies of EUS-to-EUS OKD releases that will cause fewer reboots of worker nodes.

**Reply by vrutkovs** [reply]
>there will be no update for the 4.10 version in the 4-stable channel

Correct

> If we want to apply some fix to the existing 4.10 (which is already the latest in 4-stable), we have to uprade to 4.11, or resort to the 4.10 nightly builds?

OKD still builds 4.10 nightlies, you can upgrade to any of them (however don't forget to mirror them as we keep them for 72hrs only)

>Is "EUS release" applicable to OKD?

No in terms of "support", but you can still use EUS Upgrade procedure for OKD
